locust_tasks/locustfile.py

```diff
@@ -166,49 +189,163 @@ def correct_address(self):
+        """
+        GET Start page
+        """
+        self.case = get_next_case()
+        with self.client.get('/en/start/', catch_response=True) as response:
+            verify_response('RequestUacSms-Start', self, response, 200, Page.START)
         
     @task(2)
-    def request_new_access_code(self):
-        self.client.get("/request-access-code")
+    def enter_postcode(self):
+        """
+        POST postcode
+        """
+        with self.client.post("/en/requests/access-code/enter-address/", {
+            'form-enter-address-postcode': self.case['postcode']
+        }, catch_response=True) as response:
+            self.address_to_select = extractAddress(response, self.case["uprn"])
+            verify_response('RequestUacSms-EnterAddress', self, response, 200, Page.SELECT_ADDRESS,
+                            self.case["postcode"])
 
     @task(3)
     def select_address(self):
-        # This code should arguable select one of the available addresses but running 
-        # with a fixed address doesn't seem to affect the success of the test
-        self.client.post("/request-access-code/select-address", {
-            'request-address-select': "{'uprn': '10023122452', 'address': hardcoded_address}"
-        })
+        """
+        POST uprn and whole address extracted as JSON from the HTML in previous task
+        """
+        logger.info("Address selected: " + self.address_to_select)
+        with self.client.post("/en/requests/access-code/select-address/", {
+            'form-select-address': self.address_to_select
+        }, catch_response=True) as response:
+            verify_response('RequestUacSms-SelectAddress', self, response, 200, Page.ADDRESS_CORRECT, self.case["addressLine1"])
 
     @task(4)
     def confirm_address(self):
-        self.client.post("/request-access-code/confirm-address", {
-            'request-address-confirmation': 'yes'
-        })
+        """
+        POST 'yes' to confirm address
+        """
+        with self.client.post("/en/requests/access-code/confirm-address/", {
+            'form-confirm-address': 'yes'
+        }, catch_response=True) as response:
+            verify_response('RequestUacSms-ConfirmAddress', self, response, 200, Page.SELECT_METHOD, "Text message")
 
     @task(5)
+    def select_method(self):
+        """
+        POST 'sms' to select text message as method of sending UACs
+        """
+        with self.client.post("/en/requests/access-code/select-method/", {
+            'form-select-method': 'sms'
+        }, catch_response=True) as response:
+            verify_response('RequestUacSms-SelectMethod', self, response, 200, Page.ENTER_MOBILE)
+
+    @task(6)
     def enter_mobile_number(self):
-        self.client.post("/request-access-code/enter-mobile", {
+        """
+        POST a mobile number
+        """
+        with self.client.post("/en/requests/access-code/enter-mobile/", {
             'request-mobile-number': '07714 330 933'
```

Comment:
This should use a randomly generated phone number. To detect a RHUI threading bug.

Reply:
ok - I'll try to do that then

Reply:
I've added a TODO for after CR-1379 is done